core/domain/user_services.py

```diff
@@ -965,7 +965,8 @@ def _save_user_settings(user_settings):
 
     # If user with the given user_id already exists, update that model
     # with the given user settings, otherwise, create a new one.
-    user_model = user_models.UserSettingsModel.get_by_id(user_settings.user_id)
+    user_model = user_models.UserSettingsModel.get_by_id(
```

Comment:
Did you mean to also change this one? Ditto for other UserSettingsModel & UserAuthDetailsModel gets in user_services_test--all that seems changed here is a reformatting. Am I missing something?

Reply:
I did replace all the function through out the directory `core/domain` but there were some failing backend tests. 
https://github.com/oppia/oppia/pull/11914#issuecomment-778539851 So I had to revert them by running the tests and Will fix this later.

Reply:
Can we change those tests to support the usage of `get`? (This will help us to have `get` in the entire core/domain/* files through one PR and fixing corresponding test will also help us understand the situation better for each such cases!)

Reply:
@DubeySandeep Yes, I would love to fix that too. But I'm not sure how to do that. Can you please help here

Reply:
According to 
https://github.com/oppia/oppia/blob/fe3e910b096907611bc4b285da9415c59e473e03/core/storage/base_model/gae_models.py#L227

This will return `None` right how to fix this

Reply:
Ok, I got it now. The main problem is with the tests.
When I'm using get(_, strict=False) for example here
https://github.com/oppia/oppia/blob/fd82f8cf6579cfa3deccd1ce55a89a75d34c6c34/core/domain/user_services_test.py#L1031-L1033

Here user_auth_details is `None` so the error I mentioned in the previous implementation was 
```
AttributeError: 'NoneType' object has no attribute 'deleted'
```
And I can fix this by using 
```python
    self.assertTrue(user_auth_details is None)
```
as a assertion.

Similarly for all other files where I've removed `get_by_id()`

Is this a correct approach to fix this?

Reply:
I think that's a reasonable replacement, though it's technically not the same thing (since ``get()`` can return ``None`` for multiple reasons--we can't be sure it's none for the correct reason).

@vojtechjelinek does asserting ``None`` here seem like a reasonable replacement for checking ``deleted`` or should we just be checking ``deleted`` as we do today? I'm guessing this fix will need to happen in a bunch of places, so this is establishing a pattern to follow.

Reply:
We should be checking `deleted`, I already discussed this with @FareesHussain, the solution is to introduce custom methods to models that require to be getable even in a deleted state.